core/src/main/java/com/lafaspot/imapnio/async/internal/ImapAsyncSessionImpl.java

```diff
@@ -317,6 +333,20 @@ public void handleIdleEvent(@Nonnull final IdleStateEvent idleEvent) {
             return;
 
         } else if (serverResponse.isTagged()) {
+            if (currentCmd.isCompressionRequested() && serverResponse.isOK()) {
```

Comment:
please add a state to indicate that the compress command is sent  and pending, and use that instead of iscompressionrequested

Reply:
Done.